ggml.h

```diff
@@ -423,6 +423,8 @@ extern "C" {
         struct ggml_tensor * grad;
         struct ggml_tensor * src[GGML_MAX_SRC];
 
+        bool visited;   // used to build graphs
```

Comment:
Since this is taking 4 bytes already, an idea is to make it `int mask;` or `int flags;` and utilize one of the bits as a "visited" flag. This will allow us in the future to make use of other bits for something else

Reply:
I think that the easiest way to do this is to turn them into a bitfield:
```C
uint32_t is_param:1;
uint32_t visited:1;  // used to build graphs
```
Adding additional flags won't increase the size of the struct.